File: microservices-todo/microservices/notification-service/services/user_service.py
import requests
import os

# User service URL
USER_SERVICE_URL = 'http://user-service:8081' if os.environ.get('FLASK_ENV') == 'production' else 'http://localhost:8081'

def get_user(user_id):
    """
    Get a user by ID from the user service
    """
    try:
        response = requests.get(f'{USER_SERVICE_URL}/api/users/{user_id}')
        if response.status_code == 200:
            return response.json()
        return None
    except Exception as e:
        print(f"Error retrieving user: {e}")
        return None

File: microservices-todo/microservices/task-service/services/userService.js
const axios = require("axios");

// User service URL (use localhost in development)
const USER_SERVICE_URL =
  process.env.NODE_ENV === "production"
    ? "http://user-service:8081"
    : "http://localhost:8081";

// TODO-COMM1: Implémentez la fonction checkUserExists
// Cette fonction doit envoyer une requête GET au service Utilisateurs
// pour vérifier si un utilisateur avec l'ID spécifié existe
// Elle doit retourner true si la requête réussit (code 200), false sinon
const checkUserExists = async (userId) => {
  try {
    const response = await axios.get(`${USER_SERVICE_URL}/api/users/${userId}`);
    return response.status === 200;
  } catch (error) {
    console.error("Error checking user existence:", error.message);
    return false;
  }
};

module.exports = {
  checkUserExists,
};
